nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -276,6 +304,9 @@ in
       { assertion = cfg.configFile == configFile -> cfg.adapter == "caddyfile" || cfg.adapter == null;
         message = "To specify an adapter other than 'caddyfile' please provide your own configuration via `services.caddy.configFile`";
       }
+      { assertion = cfg.enableReload -> !adminDisabled;
+        message = "You need to remove `admin off` from your Caddy configuration in order to use `services.caddy.enableReload`";
```

Comment:
Since users can set `services.caddy.configFile = /path/to/config.json`, "remove `admin off`" is not general enough.